src/Uno.UI.Composition/Composition/IPointerRedirector.cs

```diff
@@ -0,0 +1,12 @@
+﻿#nullable enable
+
+using System.Collections.Generic;
+using Microsoft.UI.Composition.Interactions;
+using Windows.UI.Input;
+
+namespace Microsoft.UI.Composition;
+
+internal interface IPointerRedirector
+{
+	void RedirectPointer(PointerPoint pointer, List<InteractionTracker> trackers);
```

Comment:
Why we pass a `List` as a parameter? I would let the `INputManager` handle the list itself.

Reply:
@dr1rrb I don't see a good way for `InputManager` to be able to access this list. But if you have something in mind I'll refactor the code.

Reply:
Well I for me the public API should be `void RedirectPointer(PointerPoint pointer, InteractionTracker trackers);`, as from an external point of view it's only about requesting to the redirect the pointer to a given tracker. Then it's the responsibility of the `InputManager` to handle the fact that multiple trackers can request redirection for the same pointer.

Basically it's only a question of the ownership of that `List`. I would imagine that the caller of this method as no needs to have access to the list, so it would make sense that it's the `InputManager` that creates it, add items and clear it on pointer release (and then recycle it).

Reply:
@dr1rrb The thing is that we already have it as a List. If I understand your suggestion, it means we will be creating a second list.

```csharp
foreach (var tracker in Trackers) // Trackers is VisualInteractionSource.Trackers
{
    Redirect(pp, tracker); // the tracker will be eventually sent to InputManager, which will create another list.
}
```

Reply:
But I also see a problem with the current approach. If an element got removed from `VisualInteractionSource.Trackers` *after* redirect and before manipulation ended, it will be problematic in two ways:

- The events should still be routed to that tracker
- A possible exception if this happened while we are enumerating the list (the "list was modified while being enumerated" exception thing, I don't recall the exact message)

Reply:
So yeah it seems we'll need to have a second list anyways